js-api-doc/value/calculation.d.ts

```diff
@@ -0,0 +1,139 @@
+ *
+ * @category Custom Function
+ */
+export class SassCalculation extends Value {
+  /**
+   * Creates a value that represents `calc(argument)`.
+   *
+   * @throws `Error` if `argument` is or transitively contains a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `calc` and `argument` as its single
+   * argument.
+   */
+  static calc(argument: CalculationValue): SassCalculation;
+
+  /**
+   * Creates a value that represents `min(arguments...)`.
+   *
+   * @throws `Error` if any of `arguments` are or transitively contain a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `min` and `arguments` as its arguments.
+   */
+  static min(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassCalculation;
+
+  /**
+   * Creates a value that represents `max(arguments...)`.
+   *
+   * @throws `Error` if any of `arguments` are or transitively contain a quoted
+   * {@link SassString}
+   * @returns A calculation with the name `max` and `arguments` as its arguments.
+   */
+  static max(
+    arguments: CalculationValue[] | List<CalculationValue>
+  ): SassCalculation;
+
+  /**
+   * Creates a value that represents `clamp(value, min, max)`.
+   *
+   * @throws `Error` if any of `value`, `min`, or `max` are or transitively
+   * contain a quoted {@link SassString}.
+   * @throws `Error` if `value` is undefined and `max` is not undefined.
+   * @throws `Error` if `value` or `max` is undefined and `min` is not a
+     {@link SassString} or {@link CalculationInterpolation} that contains
+     comma-separated values that can be interpreted as values for `value` and
+     `max` (for example `clamp(#{"1, 2, 3"})`).
+     @returns A calculation with the name `clamp` and `min`, `value`, and `max`
+     as it's arguments, excluding any arguments that are undefined.
+   */
+  static clamp(
+    min: CalculationValue,
+    value?: CalculationValue,
+    max?: CalculationValue
+  ): SassCalculation;
+
+  /** Returns the calculation's `name` field. */
+  get name(): string;
+
+  /** Returns a list of the calculation's `arguments` */
+  get arguments(): List<CalculationValue>;
+}
+
+/**
+ * The set of possible operators in a Sass calculation.
+ * @category Custom Function
+ */
+export type CalculationOperator = '+' | '-' | '*' | '/';
+
+/**
+ * A binary operation that can appear in a {@link SassCalculation}.
+ * @category Custom Function
+ */
+export class CalculationOperation implements ValueObject {
+  /**
+   * Creates a Sass CalculationOperation with the given `operator`, `left`, and
+   * `right` values.
+   */
+  constructor(
+    operator: CalculationOperator,
+    left: CalculationValue,
+    right: CalculationValue
+  );
+
+  /** Returns the operation's `operator` field. */
+  get operator(): CalculationOperator;
+
+  /** Returns the operation's `left` field. */
+  get left(): CalculationValue;
+
+  /** Returns the operation's `right` field. */
+  get right(): CalculationValue;
```

Comment:
Curious as to why these are specified as getters instead of `readonly` properties?

Reply:
We want to leave open the possibility for implementations to implement these using getters. Does TypeScript allow that for readonly properties? I don't actually know.

Reply:
To test, I went with `readonly` in https://github.com/oddbird/embedded-host-node/blob/js-api-calculations/lib/src/value/calculations.ts#L113-L115 and tsc didn't complain, but I also see the value in making them getters. Will defer to you on this one

Reply:
We use getters elsewhere so I think let's stick with them for consistency.